pkg/controller/cert_control.go

```diff
@@ -0,0 +1,323 @@
+		if err != nil {
+			return nil, fmt.Errorf("failed to delete exist old CSR for [%s/%s]: %s, error: %v", ns, instance, csrName, err)
+		}
+		glog.Infof("exist old CSR deleted for [%s/%s]: %s", ns, instance, csrName)
+		return rcc.sendCSR(ns, instance, rawCSR, csrName)
+	}
+
+	csr = &capi.CertificateSigningRequest{
+		TypeMeta: types.TypeMeta{Kind: "CertificateSigningRequest"},
+		ObjectMeta: types.ObjectMeta{
+			Name:   csrName,
+			Labels: make(map[string]string),
+		},
+		Spec: capi.CertificateSigningRequestSpec{
+			Request: pem.EncodeToMemory(&pem.Block{
+				Type:    "CERTIFICATE REQUEST",
+				Headers: nil,
+				Bytes:   rawCSR,
+			}),
+			Usages: []capi.KeyUsage{
+				capi.UsageClientAuth,
+				capi.UsageServerAuth,
+			},
+		},
+	}
+
+	labelTemp := label.New()
+	csr.Labels[label.NamespaceLabelKey] = ns
+	csr.Labels[label.ManagedByLabelKey] = labelTemp[label.ManagedByLabelKey]
+	csr.Labels[label.InstanceLabelKey] = instance
+
+	resp, err := rcc.kubeCli.CertificatesV1beta1().CertificateSigningRequests().Create(csr)
+	if err != nil {
+		return resp, fmt.Errorf("failed to create CSR for [%s/%s]: %s, error: %v", ns, instance, csrName, err)
+	}
+	glog.Infof("CSR created for [%s/%s]: %s", ns, instance, csrName)
+	return resp, nil
+}
+
+func (rcc *realCertControl) approveCSR(csr *capi.CertificateSigningRequest) error {
+	csr.Status.Conditions = append(csr.Status.Conditions, capi.CertificateSigningRequestCondition{
+		Type:    capi.CertificateApproved,
+		Reason:  "AutoApproved",
+		Message: "Auto approved by TiDB Operator",
+	})
+	_, err := rcc.kubeCli.CertificatesV1beta1().CertificateSigningRequests().UpdateApproval(csr)
+	if err != nil {
+		return fmt.Errorf("error updating approval for csr: %v", err)
+	}
+	return nil
+}
+
+/*
+func (rcc *realCertControl) RevokeCert() error {
+	return nil
+}
+*/
+/*
+func (rcc *realCertControl) RenewCert() error {
+	return nil
+}
+*/
+
+// LoadFromSecret loads cert and key from Secret matching the name
+func (rcc *realCertControl) LoadFromSecret(ns string, secretName string) ([]byte, []byte, error) {
+	secret, err := rcc.kubeCli.CoreV1().Secrets(ns).Get(secretName, types.GetOptions{})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return secret.Data["cert"], secret.Data["key"], nil
+}
+
+func (rcc *realCertControl) SaveToSecret(ns string, instance string, component string, suffix string, cert []byte, key []byte) error {
+	secretName := fmt.Sprintf("%s-%s", instance, suffix)
+
+	secret := &corev1.Secret{
+		ObjectMeta: types.ObjectMeta{
+			Name:   secretName,
+			Labels: make(map[string]string),
+		},
+		Data: map[string][]byte{
+			"cert": cert,
+			"key":  key,
+		},
+	}
+
+	labelTemp := label.New()
+	secret.Labels[label.NamespaceLabelKey] = ns
+	secret.Labels[label.ManagedByLabelKey] = labelTemp[label.ManagedByLabelKey]
+	secret.Labels[label.InstanceLabelKey] = instance
+	secret.Labels[label.ComponentLabelKey] = component
+
+	_, err := rcc.kubeCli.CoreV1().Secrets(ns).Create(secret)
```

Comment:
In order to keep the code style consistent, I think need to create a `secret_control.go` to manage the secret-related operations like [service_control.go](https://github.com/pingcap/tidb-operator/blob/master/pkg/controller/service_control.go) does.